File: algo/sts3006/Sieve.java
package sts3007; 

class Sieve {
    void simpleSieve(int n)
    {
        boolean prime[] = new boolean[n+1]; 
        for (int i = 0; i < n; i++) { 
            prime[i] = true;
        }

        for (int p = 2; p * p <= n; p++) { 
            if (prime[p]) {
                for (int i = p * p; i <= n; i += p) {
                    prime[i] = false; 
                }
            }
        }

        for (int i = 2; i <= n; i++) { 
            if (prime[i]) { 
                System.out.print(i + " "); 
            }
        }
    }
    public static void main(String[] args) {
        int n = 50; 
        Sieve sieve = new Sieve(); 
        sieve.simpleSieve(n);
    }
}

File: algo/sieve.cpp
#include <iostream>
#include <vector>

using namespace std;

void sieve(int n) {
    vector<bool> prime(n+1, true); 
    for (int i = 2; i * i <=n; i++) { 
        if (prime[i]) { 
            for (int j = 2 * i; j <= n; j += i) {
                prime[i] = false; 
            }
        }
    }

    for (int i = 2; i <= n; i++) { 
        if (prime[i]) {
            cout << i << " "; 
        }
    }
}

int main() { 
    int n = 77; 
    sieve(n); 
}